src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -518,9 +518,20 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
                         // track references for a large number of feeds (e.g., every feed in New
                         // York State).
                         if (mergeType.equals(MTC)) {
-                            Set<NewGTFSError> idErrors = referenceTracker
-                                .checkReferencesAndUniqueness(keyValue, lineNumber, field, val,
-                                    table, keyField, orderField);
+                            Set<NewGTFSError> idErrors;
+                            // If analyzing the second feed (non-future feed), the service_id always gets feed scoped.
+                            // See https://github.com/ibi-group/datatools-server/issues/244
+                            if (feedIndex == 1 && field.name.equals("service_id")) {
+                                valueToWrite = String.join(":", idScope, val);
```

Comment:
I think this feed-scoped value needs to get updated in mergedFeedsResult:
```
mergeFeedsResult.remappedIds.put(key, valueToWrite);
```

Reply:
Hmmm, now that you mention that, it looks like there could be a bug that happens on remapping calendar IDs of a feed that only has calendar_dates. I'm going to make a test case and fix for that.

Reply:
Well, after quite a bit of investigation and more testing I did add it the code you mentioned, but the effect of doing that is nearly inconsequential when there is a feed that only has calendar_dates. I wrote a test case for this and since my code will scope any service_id for any field, this additional check and renaming doesn't happen for calendar_dates. Regardless, the resulting merged feed is technically correct and passes the test case. However, the MergeFeedResult#remappedReferences is not incremented properly when the scoped feed has only calendar_dates. That metric is displayed in the UI, but I'm wondering do we really need to show it?

Additionally, I noticed what might be 2 other problems:

1. Odd skipping logic

In the calculations for whether a service_id in an active feed should be skipped, the comparison of this is done against the current day, not the start day of the future feed. This seems odd. It seems like it would make more sense to have the comparison date be the start date of the future feed and then skip various service_id's in the active feed if they occur after that date.

2. calendar_dates skipping

There is currently no logic for skipping service_ids that occur in the calendar_dates file. Shouldn't there also be logic in this area?

Reply:
Waiting on response for item 2 from MTC. Both items currently captured in #256.